casts/cards/src/UICollectionCard/CollectionSlide.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { View, StyleSheet, LayoutChangeEvent, LayoutRectangle } from 'react-native';
+import { UIImage, UIVideo } from '@tonlabs/uikit.media';
+
+import type { CollectionSlideProps } from './types';
+
+const defaultDimension: LayoutRectangle = {
+    x: 0,
+    y: 0,
+    width: 0,
+    height: 0,
+};
+
+export function CollectionSlide({ content, style }: CollectionSlideProps) {
+    const [dimensions, setDimensions] = React.useState<LayoutRectangle>(defaultDimension);
+    const onLayout = (event: LayoutChangeEvent) => {
+        setDimensions(event.nativeEvent.layout);
+    };
```

Comment:
So, if a content is an Image we allocate unnecessary `state` and create onLayout callback, even though it's not used?

Reply:
Fixed. Moved to separate component.